### ISSUE TITLE
Error trying to read http://www.shamusyoung.com/diecast/mumblo.mp3

### ISSUE THREAD

**Issue by bwinton** [agersant/mp3-duration#2]
(It also fails if I download the file locally and run:
```rust
  let path = Path::new("mumblo.mp3");
  let duration = mp3_duration::from_path(&path).unwrap();
```
)

The error is:
```
$ RUST_BACKTRACE=1 cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.2 secs
     Running `target/debug/podcasts`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: UnexpectedEof, error: StringError("failed to fill whole buffer") }) }

stack backtrace:
   0:     0x7ff1ada1dc6c - backtrace::backtrace::libunwind::trace
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.5/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.5/src/backtrace/mod.rs:42
   1:     0x7ff1ada17d2f - backtrace::capture::{{impl}}::new_unresolved
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.5/src/capture.rs:88
   2:     0x7ff1ada16988 - failure::backtrace::internal::{{impl}}::new
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/failure-0.1.1/src/backtrace/internal.rs:44
   3:     0x7ff1ada14fee - failure::backtrace::{{impl}}::new
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/failure-0.1.1/src/backtrace/mod.rs:98
   4:     0x7ff1ada0ec37 - failure::error::{{impl}}::from<std::io::error::Error>
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/failure-0.1.1/src/error.rs:34
   5:     0x7ff1ada0c2a6 - mp3_duration::from_read<std::io::buffered::BufReader<&std::fs::File>>
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/mp3-duration-0.1.2/src/lib.rs:196
   6:     0x7ff1ada0d1d3 - mp3_duration::from_file
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/mp3-duration-0.1.2/src/lib.rs:269
   7:     0x7ff1ada09de3 - mp3_duration::from_path<&&std::path::Path>
                        at /home/bwinton/.cargo/registry/src/github.com-1ecc6299db9ec823/mp3-duration-0.1.2/src/lib.rs:288
   8:     0x7ff1ada09e9b - podcasts::main
                        at src/main.rs:94
   9:     0x7ff1ada48c7e - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  10:     0x7ff1ada40bfb - std::panicking::try<(),closure>
                        at /checkout/src/libstd/panicking.rs:459
                         - std::panic::catch_unwind<closure,()>
                        at /checkout/src/libstd/panic.rs:365
                         - std::rt::lang_start
                        at /checkout/src/libstd/rt.rs:58
  11:     0x7ff1ada09f6d - main
  12:     0x7ff1ac9f082f - __libc_start_main
  13:     0x7ff1ada08ab8 - _start
  14:                0x0 - <unknown>', /checkout/src/libcore/result.rs:906:4
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:68
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:57
             at /checkout/src/libstd/panicking.rs:381
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:397
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:577
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:538
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:522
   7: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:498
   8: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:71
   9: core::result::unwrap_failed
             at /checkout/src/libcore/macros.rs:23
  10: <core::result::Result<T, E>>::unwrap
             at /checkout/src/libcore/result.rs:772
  11: podcasts::main
             at src/main.rs:94
  12: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:99
  13: std::rt::lang_start
             at /checkout/src/libstd/panicking.rs:459
             at /checkout/src/libstd/panic.rs:365
             at /checkout/src/libstd/rt.rs:58
  14: main
  15: __libc_start_main
  16: _start
```

**Comment by agersant** [comment]
Took a while but I got to the bottom of this. It turns out the file is truncated, it is missing some audio data at the end (most likely a fraction of a second).

I debated whether I wanted to support this or not. Technically that file isn't a valid mp3 file, but I think it's fair to measure the data which is actually present in the file. Media players seem to support this and it is hopefully more useful to end users.

Fixed in version 0.1.3.
